provider/snowflake.go

```diff
@@ -123,138 +125,110 @@ func (sf *snowflakeOfflineStore) UpdateTransformation(config TransformationConfi
 	return fferr.NewInternalErrorf("Snowflake Offline Store does not currently support updating transformations")
 }
 
-// RegisterResourceFromSourceTable creates a new table in the database for labels, but not features. Unlike the sqlOfflineStore implementation, which
-// creates a resource table for features, there's no need here for the intermediate table between source transformation and materialization table.
+// Snowflake breaks with the pattern of other offline store that create resource tables for labels and features. (Resource tables are intermediate tables
+// between the sources on which labels and features are registered and materializations and training sets; they duplicate the data from the source tables
+// for the 2 columns provided, that is, entity, value, and timestamp.)
 func (sf *snowflakeOfflineStore) RegisterResourceFromSourceTable(id ResourceID, schema ResourceSchema, opts ...ResourceOption) (OfflineTable, error) {
-	logger := sf.logger.With("id", id, "schema", schema, "opts", opts)
-	logger.Info("Snowflake offline store creating resource table...")
-	if id.Type == Feature {
-		logger.Debugw("Snowflake Offline Store does not support creating resource tables for features")
-		return nil, nil
+	if len(opts) > 0 {
+		return nil, fferr.NewInvalidArgumentErrorf("snowflake offline store does not currently support resource options")
 	}
-	if err := id.check(Label); err != nil {
-		logger.Errorw("Failed to validate resource ID", "error", err)
+	containsCols, err := sf.checkSourceContainsResourceColumns(id, schema, opts...)
+	if err != nil {
 		return nil, err
 	}
-	if len(opts) == 0 || opts[0].Type() != SnowflakeDynamicTableResource {
-		logger.Errorw("Snowflake Offline Store requires Dynamic Table Resource Options")
-		return nil, fferr.NewInternalErrorf("Snowflake Offline Store requires Dynamic Table Resource Options")
-	}
-	opt, isSnowflakeConfigOpt := opts[0].(*ResourceSnowflakeConfigOption)
-	if !isSnowflakeConfigOpt {
-		logger.Errorw("Snowflake Offline Store requires Resource Config Options; received %T", opts[0])
-		return nil, fferr.NewInternalErrorf("Snowflake Offline Store requires Resource Config Options; received %T", opts[0])
-	}
-	tableConfig := opt.Config
-	if tableConfig == nil {
-		logger.Debugw("Resource table config for label resource is empty")
-		tableConfig = &metadata.SnowflakeDynamicTableConfig{}
-	}
-	logger.Debugw("Resource table config for label resource before Merge with Snowflake Config", "config", tableConfig)
-	resConfig := &metadata.ResourceSnowflakeConfig{
-		DynamicTableConfig: tableConfig,
-		Warehouse:          opt.Warehouse,
-	}
-	var snowflakeConfig pc.SnowflakeConfig
-	if err := snowflakeConfig.Deserialize(sf.sqlOfflineStore.Config()); err != nil {
-		logger.Errorw("Failed to deserialize snowflake config", "error", err)
-		return nil, err
+	if !containsCols {
+		return nil, fferr.NewInvalidArgumentErrorf("source table does not contain expected columns")
```

Comment:
Would be nice to know which column isn't here

Reply:
Actually the way your code works the error returned is this error with the columns attached maybe don't do it that way and just return the diff?

Reply:
Just changed the return type to `[]string` such that the diff if columns between the actual and expected can be returned and logged.